File: utils/math/vee.m
function vec = vee(ss)
% function vec = vee(skewsymmetric)
%   The vee function maps a skew-symmetric 
%   matrix to a vector

if isa(ss,'sym')
    ss = expand(simplify(ss));
end

switch numel(ss)
    
    case 4
        
        if ~isequal(ss(1,2), -ss(2,1))
            warning('The provided matrix is not skew symmetric')
        end
        
        vec = ss(1,2);

    case 9

        if ~isequal(ss(3,2),-ss(2,3)) || ~isequal(ss(1,3),-ss(3,1)) || ~isequal(ss(2,1),-ss(1,2))
            warning('The provided matrix is not skew symmetric.')
        end

        vec = [ss(3,2); ss(1,3); ss(2,1)];
end

if isa(ss,'sym')
    vec = simplify(vec);
end

end

File: utils/math/vee.py
import numpy as np

def vee(ss):
    size = ss.shape[0]*ss.shape[1]
    if size == 4:
        if (ss[0,1]==ss[1,0]):
            vec = ss[0,1]
    elif size == 9:
        if (ss[2,1]==-ss[1,2] and ss[0,2]==-ss[2,0] and ss[1,0==-ss[0,1]]):
            vec = np.array([[ss[2,1]],[ss[0,2]],[ss[1,0]]])
    return vec